src/test/java/io/stargate/sgv2/jsonapi/api/v1/RangeReadIntegrationTest.java

```diff
@@ -0,0 +1,419 @@
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.document", is(notNullValue()))
+          .body("data.document", jsonEquals(expected));
+    }
+
+    @Test
+    @Order(11)
+    public void gtWithDeleteOne() throws Exception {
+      String json =
+          """
+        {
+          "deleteOne": {
+            "filter" : {"userId" : {"$gt" : 23}},
+            "sort" : {"userId" : 1}
+          }
+        }
+        """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("errors", is(nullValue()))
+          .body("status.deletedCount", is(1));
+    }
+
+    @Test
+    @Order(12)
+    public void gtWithDeleteMany() throws Exception {
+      String json =
+          """
+              {
+                "deleteMany": {
+                  "filter" : {"userId" : {"$gte" : 23}}                }
+              }
+              """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("errors", is(nullValue()))
+          .body("status.deletedCount", is(2));
+    }
+  }
+
+  private List<Object> getDocuments(int countOfDocuments) {
+    List<Object> data = new ArrayList<>(countOfDocuments);
+    for (int docId = 1; docId <= countOfDocuments; docId++) {
+      data.add(
+          new TestData(
+              docId,
+              "user" + docId,
+              docId,
+              docId % 2 == 0,
+              new DateValue(1672531200000L + docId * 1000)));
+    }
+    return data;
+  }
+
+  private List<Object> getDocuments(int[] docIds) {
+    List<Object> data = new ArrayList<>(docIds.length);
+    for (int docId : docIds) {
+      data.add(
+          new TestData(
+              docId,
+              "user" + docId,
+              docId,
+              docId % 2 == 0,
+              new DateValue(1672531200000L + docId * 1000)));
+    }
+    return data;
+  }
+
+  private void insert(List<Object> testDatas) {
+    testDatas.forEach(
+        testData -> {
+          String json = null;
+          try {
+            json = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(testData);
+          } catch (JsonProcessingException e) {
+            e.printStackTrace();
```

Comment:
Rather throw something like `IllegalArgumentException`, to make sure test fails if this ever was to happen